cpp/src/arrow/flight/client.h

```diff
@@ -41,6 +42,29 @@ namespace flight {
 
 class ClientAuthHandler;
 
+/// \brief A duration type for Flight call timeouts.
+typedef std::chrono::duration<double, std::chrono::seconds::period> TimeoutDuration;
+
+/// \brief Hints to the underlying RPC layer for Arrow Flight calls.
```

Comment:
If this is simply a container for configuration values, perhaps we can make it a light-weight `struct` with public fields and no implementation class?

Reply:
That makes sense.